packages/angular_devkit/build_angular/src/tools/babel/plugins/add-code-coverage.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { NodePath, PluginObj, types } from '@babel/core';
+import { Visitor, programVisitor } from 'istanbul-lib-instrument';
+import assert from 'node:assert';
+
+/**
+ * A babel plugin factory function for adding istanbul instrumentation.
+ *
+ * @returns A babel plugin object instance.
+ */
+export default function (): PluginObj {
+  const visitors = new WeakMap<NodePath, Visitor>();
+
+  return {
+    visitor: {
+      Program: {
+        enter(path, state) {
+          const visitor = programVisitor(types, state.filename, {
```

Comment:
- The exclusion of `options.instrumentCode.includedBasePath` leads to instrumenting files outside the project scope.
- To maintain the current functionality, `node_modules` should also be excluded from instrumentation.

Reply:
That's handled here:
https://github.com/angular/angular-cli/blob/6c6985489f24ee820a9b08c9d09b78bd4ea18f3d/packages/angular_devkit/build_angular/src/tools/babel/webpack-loader.ts#L150-L155